---
title: "rmsfasta"
author: Jeff Jones
output:
  md_document:
    variant: markdown_github
---

```{r, echo = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "README-"
)
```

An lightweight R package parsing [FASTA](https://en.wikipedia.org/wiki/FASTA_format) (like those from [UniProt](https://www.uniprot.org/)) files into an R usable `list` or `data.frame`. The main function utilizes [regular expressions](https://en.wikipedia.org/wiki/Regular_expression) to extract meta data along with the protein sequence.

## Installation

To install, open R and type:

``` r
install.packages("devtools")
devtools::install_github("jeffsocal/rmsfasta")
```

## Get Started

Its simple to get started, just point the `read_fasta()` function at a downloaded FASTA file and save as an object.

``` {r}
library(rmsfasta)

path_to_fasta <- system.file("extdata", "albu_human.fasta", package = "rmsfasta")
fasta_data <- read_fasta(path_to_fasta)

fasta_data[[1]]
```

In addition, the expected peptides can be generated with the `digest()` function.
``` {r}
fasta_data <- lapply(fasta_data, digest)

fasta_data[[1]]
```

Alternatively read_fasta the FASTA file into a `data.frame`.
``` {r}
library(tidyverse, warn.conflicts = FALSE)
fasta_data <- read_fasta(path_to_fasta, as = "data.frame") |> as_tibble()

fasta_data
```
